### DISCUSSION TITLE
LILYGO TTGO T-Beam M8N & SX1262 won't update

### DISCUSSION THREAD

**Discussion by ricknun** [meshtastic/Discussions#79] [ESP32]
I have a LILYGO TTGO T-Beam M8N & SX1262 purchased in 2022.  It agrees with the board photo shown here:
https://meshtastic.org/docs/hardware/devices/lilygo/tbeam/?t-beam=sx1262

Today it powered on fine with firmware 1.2.50.  The display shows "1/1 Fixed GPS #LongSlow-N".  In 2022 I used the Windows CLI and python to program it.  Today on my Windows 10 PC I used Google Chrome and Meshtastic Web Flasher to update it with **Device = LILYGO T-Beam**, Firmware = 2.5.11.8e2a3e5, Flash, COM15, Connect.  I see "Flashing App Partition xx%" up to 100% (good so far) but then a forever-blinking 2 times per second text display ending in what is posted below.  Then the device is dead.  After programming I've tried disconnecting USB and connecting power; RST; press/hold power; hold down RST and power-on.  **Am I using the right Web Flasher "Device" choice?  What else can I try?**

... (can't copy anything before what follows)
rst:0x3 (SW_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0018,len:4
load:0x3fff001c,len:1044
load:0x40078000,len:8896
load:0x40080400,len:5828
entry 0x400806ac

**Comment by ricknun** [comment]
No more need of an answer.  I spent the day updating and fixing problems with python (v3.13.0 is installed and v3.8 was being used), pip, pytap, esptool, and meshtastic.  Then I could use the Command Line Interface.  So easy...  (:-)

**Comment by thebentern** [comment]
Hi Rick, when you used the web-flasher, did you select the Full erase and install? If you're updating from a very old version like 1.2.X, a normal firmware update is more likely to fail.

**Comment by ricknun** [comment]
Thank you for the reply, but I want to comment for other readers -- I didn't see anything in my use of Web Flasher to indicate when "update" would and wouldn't work, or that the failure display I described (infinite looping of unclear text) meant "It won't work in your case and you must do a full erase and install".  I think that was the root cause of my problems and what wasted hours of time.  It appears to work, programming the device and slowly counting up percentage done, then loops forever.  Better documentation and failure indications would have really helped me.
